packages/clerk-js/src/ui/elements/withAvatarShimmer.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import { useAppearance } from '../customizables';
+import { ThemableCssProp } from '../styledSystem';
+
+export const withAvatarShimmer = <T extends { sx?: ThemableCssProp }>(
+  Component: React.ComponentType<T>,
+): React.ComponentType<T> => {
+  return (props: T) => {
+    const { parsedLayout } = useAppearance();
+
+    return (
+      <Component
+        {...props}
+        sx={[
+          t =>
+            parsedLayout.avatarShimmer && {
+              ':hover': {
+                '--cl-shimmer-hover-shadow': `1px 1px 2px ${t.colors.$blackAlpha500}`,
+                '--cl-shimmer-hover-transform': 'skew(-45deg) translateX(600%)',
```

Comment:
nice

Reply:
IIRC  we need this HOC because we want the effect to be triggered when hovering a parent element that may not completely overlap with the avatar, eg when the parent container is wider than the avatar (org switcher case)

Is that correct? If so, please add a comment in the code :)

Reply:
Yes, that is correct.